rein/cli.py

```diff
@@ -1576,6 +1584,65 @@ def unhide():
         except:
             return 'false'
 
+    @app.route('/config', methods=['POST'])
+    def config_web():
+        """Allows for changes to the user's config via the web interface"""
+
+        try:
+            key = request.json['key']
+            value = request.json['value']
+            config_common(key, value)
+
+        except:
+            return 'false'
+        
+        return 'true'
+
+    @app.route('/trust_score/<dest_msin>', defaults={'source_msin': user.msin})
+    @app.route('/trust_score/<dest_msin>/<source_msin>', methods=['GET'])
+    def trust_score(dest_msin, source_msin):
+        """Calculates the trust score for a user as identified by his msin.
+        Algorithm based on the level 2 trust system implemented by Bitcoin OTC
+        and outlined at https://wiki.bitcoin-otc.com/wiki/OTC_Rating_System#Notes_about_gettrust."""
+
+        # Grab all ratings commited by source_msin (the client)
+        ratings_by_source = rein.session.query(Document).filter(and_(
+            Document.doc_type == 'rating',
+            Document.contents.like('%\nRater msin: {}%'.format(source_msin))
+        )).all()
+
+        # Compile list of users (by msin) that have been rated by source and their ratings
+        rated_by_source = []
+        for rating in ratings_by_source:
+            rating_dict = document_to_dict(rating.contents)
+            msin_rated = rating_dict['User msin']
+            rating_value = int(rating_dict['Rating'])
+            if not msin_rated in rated_by_source:
+                rated_by_source.append((msin_rated, rating_value))
+
+        # Determine if any of the users rated by source or source have rated dest
+        # Add source to the list of users source has vouched for with full trust
+        vouched_users = rated_by_source + [(source_msin, 5)]
+        # Calculate trust links between source and dest
+        trust_links = []
+        for vouched_user in vouched_users:
```

Comment:
It currently doesn't follow more than two links because I didn't see that implemented in bitcoin otc.

Would you like that? If yes, we'd probably need a hard limit for how many links it should follow at most. Cyclic ratings could probably easily be avoided by keeping track of all source msins. 

Some tests definitely seem sensible, I'll think of something. That example is fairly simple though. Because chains are one-sided there is only one link (B) between A and D, giving D a trust score of 4.